ocf_datapipes/load/satellite.py

```diff
@@ -13,40 +13,83 @@
 _log = logging.getLogger(__name__)
 
 
-def open_sat_data(zarr_path: Union[Path, str, list[Union[str, Path]]]) -> xr.DataArray:
+def _get_single_sat_data(zarr_path: Union[Path, str]) -> xr.DataArray:
+    """Helper function to open a zarr from either local or GCP path.
+
+    The local or GCP path may contain wildcard matching (*)
+
+    Args:
+        zarr_path: Path to zarr file
+    """
+
+    # These kwargs are used if zarr path contains "*"
+    openmf_kwargs = dict(
+        engine="zarr",
+        concat_dim="time",
+        combine="nested",
+        chunks="auto",
+        join="override",
+    )
+
+    # Need to generate list of files if using GCP bucket storage
+    if "gs://" in str(zarr_path) and "*" in str(zarr_path):
+        result_string = subprocess.run(
```

Comment:
Is this necessary for xarray? I've definitely opened up the satellite zarrs with just the wildcard, and not needing this extra subprocess.

Reply:
To confirm, I tested this again on the non-HRV data and trying to open with:

```
dataset = xr.open_mfdataset("gs://.../*_nonhrv.zarr")
```
hangs for at least a few minutes.

Listing the files with subprocess and then opening these files completes in seconds.